dotcom-rendering/src/components/NewsletterCard.tsx

```diff
@@ -71,13 +85,61 @@ export const buttonStyle = css`
 	}
 `;
 
-export const getButtonInitialAriaAttribute = (newsletterName: string) => ({
+const haveSeenMaker = css`
+	position: absolute;
+	bottom: 30;
+	right: 0;
+`;
+
+const getButtonInitialAriaAttribute = (newsletterName: string) => ({
 	'aria-label': `add ${newsletterName} to subscribe list`,
 	'data-aria-label-when-unchecked': `add ${newsletterName} to subscribe list`,
 	'data-aria-label-when-checked': `remove ${newsletterName} from subscribe list`,
 });
 
-export const NewsletterCard = ({ newsletter }: { newsletter: Newsletter }) => {
+export const NewsletterCard = ({
+	newsletter,
+	groupTitle,
+	cardPosition,
+	carouselPosition,
+}: Props) => {
+	const [hasBeenSeen, setRefForHaveSeenMarker] = useIsInView({});
+	const [haveReportedBeingSeen, setHaveReportedBeingSeen] = useState(false);
+
+	const reportSeen = useCallback(() => {
+		const record = getOphanRecordFunction();
+		const valueData = {
+			id: newsletter.identityName,
+			carousel: groupTitle,
+			cardPosition,
+			carouselPosition,
+		};
+
+		submitComponentEvent(
+			{
+				component: {
+					componentType: 'NEWSLETTER_SUBSCRIPTION',
+					id: `DCR NewsletterCard ${newsletter.identityName}`,
+				},
+				action: 'VIEW',
+				value: JSON.stringify(valueData),
+			},
+			record,
+		);
+	}, [cardPosition, carouselPosition, groupTitle, newsletter.identityName]);
+
+	useEffect(() => {
+		if (hasBeenSeen && !haveReportedBeingSeen) {
```

Comment:
could possibly use `useOnce` here and not need a `haveReportedBeingSeen` state, but you'd need to set the initial state of `hasBeenSeen` to undefined

Reply:
I feel it would make the logic a bit more obscure to use two custom hooks together in the component. If this pattern came up again, it might be worth writing a `useCallbackWhenInView` hook that tie it all together? Something like 
```typescript
{ 
    (callback:{(observedElement:HTMLElement|null):void}):
        Dispatch<SetStateAction<HTMLElement | null>> )
}
```

Reply:
Yea makes sense!